### ISSUE TITLE
is still repeating at this depth when missing $ before `(`

### ISSUE THREAD

**Issue by xNxExOx** [rust-lang/rust#101583]
Given the following code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2170c66f0413032a3b07da1066131517

```rust
macro_rules! tuple {
    (
        $($i:literal),*
    ) => { 
        ( ($i),* )
    };
}
```

The current output is:

```
error: variable 'i' is still repeating at this depth
 --> src/main.rs:5:12
  |
5 |         ( ($i),* )
  |            ^^
```

Ideally the output should look like:

```
error: variable 'i' is still repeating at this depth
 --> src/main.rs:5:12
  |
5 |         ( ($i),* )
  |            ^^
  did you mean
  |
5 |         ( $($i),* )
  |            ^
```

Theoretically `(...),*` suggest it should suggest repeating block, but I have no clue how to write such diagnostic, so feel free to say it would be bad, because it would be causing false positives.
